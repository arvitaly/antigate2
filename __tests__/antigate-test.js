"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const express = require("express");
const bodyParser = require("body-parser");
const __1 = require("./..");
const port = 34578;
const base64Data = "/9j/4AAQSkZJRgABAQAAAQABAAD";
let server;
let createTaskFn;
let getTaskResultFn;
beforeEach((done) => {
    const app = express();
    app.use(bodyParser.json());
    createTaskFn = jest.fn();
    getTaskResultFn = jest.fn();
    app.post("/createTask", (req, res) => {
        const result = createTaskFn(req.url, req.headers, req.body);
        res.send(JSON.stringify(result));
    });
    app.post("/getTaskResult", (req, res) => {
        const result = getTaskResultFn(req.url, req.headers, req.body);
        res.send(JSON.stringify(result));
    });
    server = app.listen(port, done);
});
it("when getByBase64, should call server with params", () => __awaiter(this, void 0, void 0, function* () {
    const key = "112";
    const antigate = new __1.Antigate({
        key,
        baseUrl: "http://127.0.0.1:" + port,
    });
    createTaskFn.mockImplementation((url, _, body) => {
        expect(url).toMatchSnapshot();
        expect(body).toMatchSnapshot();
        return {
            taskId: 111,
        };
    });
    let attemptCount = 0;
    getTaskResultFn.mockImplementation((url, _, body) => {
        expect(url).toMatchSnapshot();
        expect(body).toMatchSnapshot();
        return {
            status: attemptCount++ ? "ready" : "processing",
            solution: {
                text: "Hello",
            },
        };
    });
    const result = yield antigate.getByBase64("data:image/jpeg;base64," + base64Data);
    expect(result).toBe("Hello");
}));
it("when nocaptcha, should call server with params", () => __awaiter(this, void 0, void 0, function* () {
    const key = "112";
    const antigate = new __1.Antigate({
        key,
        baseUrl: "http://127.0.0.1:" + port,
    });
    createTaskFn.mockImplementation((url, _, body) => {
        expect(url).toMatchSnapshot();
        expect(body).toMatchSnapshot();
        return {
            taskId: 112,
        };
    });
    let attemptCount = 0;
    getTaskResultFn.mockImplementation((url, _, body) => {
        expect(url).toMatchSnapshot();
        expect(body).toMatchSnapshot();
        return {
            status: attemptCount++ ? "ready" : "processing",
            solution: {
                gRecaptchaResponse: "cpatchId",
            },
        };
    });
    const result = yield antigate.getNoCaptcha("test1", "test2");
    expect(result).toMatchSnapshot();
}));
afterEach((done) => {
    server.close(done);
});
