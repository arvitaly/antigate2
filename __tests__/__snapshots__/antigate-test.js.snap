// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when getByBase64, should call server with params 1`] = `"/createTask"`;

exports[`when getByBase64, should call server with params 2`] = `
Object {
  "clientKey": "112",
  "languagePool": "rn",
  "task": Object {
    "body": "/9j/4AAQSkZJRgABAQAAAQABAAD",
    "type": "ImageToTextTask",
  },
}
`;

exports[`when getByBase64, should call server with params 3`] = `"/getTaskResult"`;

exports[`when getByBase64, should call server with params 4`] = `
Object {
  "clientKey": "112",
  "taskId": 111,
}
`;

exports[`when getByBase64, should call server with params 5`] = `"/getTaskResult"`;

exports[`when getByBase64, should call server with params 6`] = `
Object {
  "clientKey": "112",
  "taskId": 111,
}
`;

exports[`when nocaptcha, should call server with params 1`] = `"/createTask"`;

exports[`when nocaptcha, should call server with params 2`] = `
Object {
  "clientKey": "112",
  "languagePool": "en",
  "task": Object {
    "type": "NoCaptchaTaskProxyless",
    "websiteKey": "test2",
    "websiteURL": "test1",
  },
}
`;

exports[`when nocaptcha, should call server with params 3`] = `"/getTaskResult"`;

exports[`when nocaptcha, should call server with params 4`] = `
Object {
  "clientKey": "112",
  "taskId": 112,
}
`;

exports[`when nocaptcha, should call server with params 5`] = `"/getTaskResult"`;

exports[`when nocaptcha, should call server with params 6`] = `
Object {
  "clientKey": "112",
  "taskId": 112,
}
`;

exports[`when nocaptcha, should call server with params 7`] = `
Object {
  "gRecaptchaResponse": "cpatchId",
}
`;
